shared/chat/hud/index.js

```diff
@@ -87,16 +87,19 @@ type MentionHudProps = {
 }
 
 // TODO figure typing out
-const mapStateToProps: MapStateToProps<*, *, *> = (state: TypedState, {userIds, selectedIndex, filter}) => ({
+const mapStateToProps: MapStateToProps<*, *, *> = (state: TypedState, {users, selectedIndex, filter}) => ({
   following: state.config.following,
   you: state.config.username || '',
-  data: userIds
+  data: users
     .map((u, i) => ({
-      username: u,
-      fullName: '', // TODO
-      key: u,
+      username: u.username,
+      fullName: u.fullName,
+      key: u.username,
     }))
-    .filter(u => u.username.indexOf(filter) >= 0)
+    .filter(u => {
+      filter = filter.toLowerCase()
```

Comment:
this should be hoisted to the very top of this if you're going to do it. its not strictly necessary (i'm sure this isn't a perf hot spot) but the original comment was that it gets called over and over in this map

Reply:
the other option is to lowercase it in reselect or mapstatetoprops